channeldb/kvdb/backend.go

```diff
@@ -95,6 +106,21 @@ func compactAndSwap(cfg *BoltBackendConfig) error {
 	sourceFilePath := filepath.Join(cfg.DBPath, sourceName)
 	tempDestFilePath := filepath.Join(cfg.DBPath, DefaultTempDBFileName)
 
+	// Let's find out how long ago the last compaction of the source file
+	// occurred and possibly skip compacting it again now.
+	lastCompactionDate, err := lastCompactionDate(sourceFilePath)
+	if err != nil {
+		return fmt.Errorf("cannot determine last compaction date of "+
+			"source DB file: %v", err)
+	}
+	compactAge := time.Since(lastCompactionDate)
+	if cfg.AutoCompactMinAge != 0 && compactAge <= cfg.AutoCompactMinAge {
+		log.Infof("Not compacting database file at %v, it was only "+
+			"compacted %v ago", sourceFilePath,
+			compactAge.Truncate(time.Second))
```

Comment:
depending on whether we choose hours/weeks for the default, maybe just print the full timestamp?

Reply:
Yes, makes sense. Added all relevant information to the message, might be a bit verbose now, but I like it:
```
2020-11-13 10:21:06.537 [INF] CHDB: Not compacting database file at /home/guggero/.lnd-dev-zane/data/graph/regtest/channel.db, it was last compacted at 2020-11-13 10:20:32.162557128 +0100 CET (34s ago), min age is set to 2m0s
```